### ISSUE TITLE
Crafting from dank/null inventory

### ISSUE THREAD

**Issue by caighan** [p455w0rd/DankNull#31]
Would be a great feature to have the ability to craft using the inventory of your dank/null. It seems one of the most common things you pick up is cobblestone and that's used in a lot of recipes like pistons and such. It is okay to have to pull a stack out, but still would be nice not to need to do that.

**Comment by p455w0rd** [comment]
so i figure i might give a little context..I love this idea..and since I already added forge item caps functionality so I can support better builders wands, this shouldn't be _too_ difficult for most recipe situations...might even PR to JEI. I'm assuming by "crafting from /dank/null inventory" you mean via the JEI `+` button that takes items from inventory and places them in a crafting grid? If so, this would actually be well suited for JEI itself to check all items for implementation of IInventory/IItemHandler...let me think on this a bit, because I could also add functionality to remove items from things like chest inventory when a chest is in inventory (itemstack form), and conforms to vanilla's `BlockEntityTag` standard..This could just become a mod on its own as this is something that would be VERY useful in far more situations than just with /dank/null

**Comment by p455w0rd** [comment]
please confirm this is what you are asking..if not it'll prolly get pushed to the update after next

**Comment by caighan** [comment]
Yes that sounds pretty much like what i iwant. The JEI+ especially as its annoying to have to pause open it up and pull cobble out for a piston or the like. Really appreciate your effort.

**Comment by p455w0rd** [comment]
just figured I'd drop an update..this is the issue I'm currently working on...there's a LOT more involved than I had figured..still I'm not giving up yet :)

**Comment by caighan** [comment]
Excellent Thanks!

**Comment by Rixxaw** [comment]
I think an easy work around for this would be a restocking feature.  Like you could put dank null in restocking mode and it would just restock from its inventory slots in your inventory that were items inside the dank null.  Like if you had 1 cobble in your inventory and you put the dank null into restocking mode it would fill that slot with 64 cobble.  then if you crafted items which used up that slot it would refill the slot as you crafted.  yes you would have to take the single cobble out but it would just keep refilling your slot.   Thats how the foresty backpacks worked.

**Comment by Hazmatik** [comment]
I like the idea of a restocking feature similar to how the void tear from the Reliquary mod works, it stores a large amount of items in it but leaves one stack of the item type in the players inventory. However, I don't think putting the whole /dank/null into restocking mode would be good but maybe as a per item option configurable from the GUI similar to the docking station extraction options

**Comment by Rixxaw** [comment]
Oh yeah thats a great idea.  Another mode for each slot that says "Restock" Only problem with that is what if you want to keep your extraction option but also have that slot restock?  Seems like it would be overly complex.  I mean you don't really need it to restock since you can use it to place so its kind of overlapping feature.  The other items which restock you can't just right click them on the ground and they place things so yeah scratch the idea.

**Comment by thelapisfox** [comment]
Been 2.5 years so just checking to see if this is ever going to be added to 1.12.x

**Comment by pixelblob** [comment]
>>.>
